tools/c7n_gcp/c7n_gcp/actions/core.py

```diff
@@ -63,24 +63,39 @@ def filter_resources(self, resources):
             )
         return resources
 
+    def include_default_project(self, resources, session):
+        updated_resources = list(resources)
+        default_project = session.get_default_project()
+        for resource in resources:
+            resource['default_project'] = default_project
```

Comment:
why is this being done? generally annotations should have a c7n: prefix

Reply:
https://cloud.google.com/dns/docs/reference/v1/managedZones/list
Please take a look at this API. The returned resources do not have any reference to project they are in, neither they contain a selfLink to extract those parameters. According to numerous existing actions, we are not supposed to specify project parameter in policies thus the default one - defined in env. variables and used to list those resources - is included there.
The prefix has been added, thank you for the tip.

Reply:
custodian is always executing in the context of a given project, so what purpose is there to annotating resources redundantly with projects.

Reply:
In general, it does, however, when an action (such as https://cloud.google.com/dns/docs/reference/v1/managedZones/delete) needs a project to be set explicitly, it is supposed to be already present in a resource's dictionary. As a rule, the project is extracted from selfLink with path_param_re, and currently we are not seeing any other way to refer to the current project from an action but include it explicitly with something like the proposed method.